src/utils/useGetMessage.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from 'react'
+import { LotusMessage } from '@glif/filecoin-message'
+import LotusRPCEngine from '@glif/filecoin-rpc-client'
+
+import { validateCID } from './validateCID'
+
+const lotusRPC = new LotusRPCEngine({
+  apiAddress:
+    process.env.NEXT_PUBLIC_LOTUS_NODE_JSONRPC ||
+    'https://calibration.node.glif.io'
+})
+
+interface UseGetMessageResult {
+  message: LotusMessage
+  loading: boolean
+  error: Error
+}
+
+export const useGetMessage = (cid: string): UseGetMessageResult => {
+  const [message, setMessage] = useState<LotusMessage | null>(null)
+  const [loading, setLoading] = useState<boolean>(false)
+  const [error, setError] = useState<Error | null>(null)
+
+  useEffect(() => {
+    setMessage(null)
+    setError(null)
+    if (validateCID(cid)) {
+      setLoading(true)
+      lotusRPC
+        .request('ChainGetMessage', { '/': cid })
+        .then((m: LotusMessage) => setMessage(m))
```

Comment:
Not sure if this actually matters, but `m` is not actually an `instanceof LotusMessage`. It should have the fields correctly capitalized, but it doesn't have any of the other prototype methods, which maybe end up causing weird errors upstream.

Can be solved by `Message.fromLotusType(m)` somewhere in this promise chain

Reply:
I see, yeah then maybe this should return a `Message` instead of a `LotusMessage`

Reply:
The problem is that the `lotusRPC` is not typed very well - this is a little wonky, but you could do something like this (since `request` func from the lotus RPC package takes an optional generic)

```
lotusRPC.request<LotusMessage>(...)
.then((m) => setMessage(Message.fromLotusType(m)))
```

Reply:
I agree it makes more sense to pass around `Message` objects instead of `LotusMessage`. I've changed the other hooks to take `Message` as input parameter as well